app/views/allocations/history.html.erb

```diff
@@ -2,13 +2,16 @@
 
 <h1 class="govuk-heading-l"><%= @prisoner.full_name %></h1>
 
-<% allocation_list(@history, @early_allocations, @email_histories).each do |prison, case_histories| %>
+<!-- @history is forward-sorted - having converted to a hash key -> list we need to convert back to a list, reverse and also reverse the list itself  -->
+<% @history.group_by { |history| @timeline.prison_episode(history.created_at) }.map { |k, v| [k, v] }.
+    sort_by { |k, v| k.start_date }.
+    reverse.map { |k, v| [k, v.reverse] }.each do |episode, case_histories| %>
```

Comment:
I thought we were going to move this into a Presenter to avoid over-complicating the View?

It feels like `@history` should really be a 'case history grouped by prison' Presenter or Enumerator rather than a plain Array – to avoid having to massage the data into the right shape within the View itself.